pkg/prometheus/server/operator.go

```diff
@@ -1027,9 +1027,25 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 		return fmt.Errorf("listing StatefulSet resources failed: %w", err)
 	}
 
+	c.updateConfigResourcesStatus(ctx, p, logger, *resources)
+
 	return nil
 }
 
+// updateConfigResourcesStatus updates the status of the selected configuration resources (serviceMonitor, podMonitor, scrapeClass and podMonitor).
+func (c *Operator) updateConfigResourcesStatus(ctx context.Context, p *monitoringv1.Prometheus, logger *slog.Logger, resources selectedConfigResources) {
+	if !c.configResourcesStatusEnabled {
+		return
+	}
+
+	if len(resources.sMons) == 0 {
+		return
+	}
+
+	configResourceSyncer := prompkg.NewConfigResourceSyncer(monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.PrometheusName), c.mclient, logger)
+	prompkg.AddServiceMonitorStatus(ctx, p, configResourceSyncer, resources.sMons)
```

Comment:
Adding to my comment on `prompkg.AddServiceMonitorStatus` I'd recommend that we iterate over the config resources in this function
```suggestion
        for _, sMon := range resources.sMons {
		err:= prompkg.UpdateServiceMonitorStatus(ctx, p, configResourceSyncer, sMon)
		...
	}
```

Reply:
done